server/container/util-java8/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -69,11 +80,28 @@ private FluentFutureStream(CompletableFuture<Stream<T>> completableFuture) {
             CompletableFutureUtil.thenComposeOnAll(completableFuture(), function));
     }
 
+    public <U> FluentFutureStream<U> thenComposeOnAllUnboxed(Function<T, CompletableFuture<Stream<U>>> function) {
+        return FluentFutureStream.of(
+            CompletableFutureUtil.thenComposeOnAll(completableFuture(), function))
+            .flatMap(u -> u);
+    }
+
+    public <U> FluentFutureStream<U> thenComposeOnAllUnboxedOptional(Function<T, CompletableFuture<Optional<U>>> function) {
```

Comment:
**flatMap** is something else (that do not involve future composition).

I would propose *thenFlatComposeOptional*